lldb/source/Plugins/Language/Swift/SwiftLanguage.cpp

```diff
@@ -856,6 +856,89 @@ SwiftLanguage::GetHardcodedSynthetics() {
           }
           return nullptr;
         });
+    g_formatters.push_back([](lldb_private::ValueObject &valobj,
+                              lldb::DynamicValueType dyn_type,
+                              FormatManager &format_manager)
+                               -> lldb::SyntheticChildrenSP {
+      // Try to recognize a Swift type imported in C++. These types have a
+      // static constexpr char field whose name is the type's Swift mangled
+      // name.
+      ProcessSP process_sp(valobj.GetProcessSP());
+      auto *swift_runtime = SwiftLanguageRuntime::Get(process_sp);
+      if (!swift_runtime)
+        return nullptr;
+
+      auto module = valobj.GetModule();
+      if (!module)
+        return nullptr;
+
+      auto gts = module->GetTypeSystemForLanguage(
+          lldb::LanguageType::eLanguageTypeSwift);
+      if (!gts)
+        return nullptr;
+
+      auto *ts = llvm::cast<TypeSystemSwift>(gts->get());
+
+      // This only makes sense for Clang types.
+      auto type = valobj.GetCompilerType();
+      auto tsc = type.GetTypeSystem().dyn_cast_or_null<TypeSystemClang>();
+      if (!tsc)
+        return nullptr;
```

Comment:
this early exit doesn't depend on any of the previous setup — should we hoist it to the top so we fail faster?

Reply:
I moved this upwards, but, as Jim pointed out, getting a value object's compiler type can be expensive, so I'm moving this back down.